aws/services/rds/resource.ftl

```diff
@@ -245,6 +260,49 @@
             }
         )
     /]
+
+    [#list links as linkId, link ]
+        [#if link?is_hash]
+            [#local linkTarget = getLinkTarget(occurrence, link, false) ]
+
+            [@debug message="Link Target" context=linkTarget enabled=false /]
+
+            [#if !linkTarget?has_content]
+                [#continue]
+            [/#if]
+
+            [#local linkTargetCore = linkTarget.Core ]
+            [#local linkTargetConfiguration = linkTarget.Configuration ]
+            [#local linkTargetResources = linkTarget.State.Resources ]
+            [#local linkTargetAttributes = linkTarget.State.Attributes ]
+            [#local linkTargetSolution = linkTargetConfiguration.Solution]
+
+            [#if (link.Role+" ")?lower_case?starts_with("publish ") ]
+                [#local linkRole = (link.Role+" ")?lower_case?keep_after("publish ")?trim ]
+                [#if linkRole?has_content]
+                    [#local linkRoles = (linkRole?split(","))![] ]
+                [#else]
+                    [#local linkRoles = [] ]
+                [/#if]
+                [@cfResource
+                    id=formatId(id, "instance", linkId)
+                    type="AWS::RDS::EventSubscription"
+                    properties=
+                        {
+                            "Enabled" : true,
+                            "SnsTopicArn" : linkTargetAttributes["ARN"],
+                            "SourceIds" : [ getReference(id) ],
+                            "SourceType" : "db-instance"
+                        } +
+                        attributeIfContent(
+                            "EventCategories",
+                            linkRoles
+                        )
+                    outputs=RDS_EVENT_OUTPUT_MAPPINGS
+                /]
+            [/#if]
```

Comment:
I think this would be better to move under a dedicated parameter under the solution configuration. There is nowhere to currently document what the role values are supposed to be so this will be really hard to actually use. 

My suggestion would be to add a new root level attribute on the db component with something like this structure 


```
{
   "SystemNotifications" : {
          "badEvents" : {
                 "Categories" : [ "<a list of categories>" ],
                 "Links" : {
                     "bigYellingTopic" : { 
                              "Tier" : "mgmt",
                              "Component" : "alerts" 
                      }
                 }
             }
}
```

The categories attribute can then be a list of acceptable event categories, so they can be documented and maanged.

Reply:
I was assuming that the majority of the time, it would be all alerts to the topic (ie role = publish - with no following comma delimited list) with the occasional filtered set (only really when you needed to split the alerts into multiple topics, presumably for security restrictions). The construct you're proposing seems to be based on the assumption that the majority of the time it will be restricted alert sets.

Do you envision the majority of the cases being a restricted set or have I missed something?